cmd/base64_decoder/decoder_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncoder(t *testing.T) {
+	t.Run("incorrect args", func(t *testing.T) {
+		for _, arg := range []string{
+			"",
+			"abc",
+		} {
+			out, err := exec.Command("go", "run", "decoder.go", arg).CombinedOutput()
+			require.NoError(t, err)
+
+			expectedOut := help + "\n  -getresponse string\n    \tjson output of " +
+				"GetDataResponseEnvelope. The value field in the json output will be decoded. " +
+				"Surround the JSON data with single quotes.\n"
+			require.Equal(t, expectedOut, string(out))
+		}
+	})
+
+	t.Run("correct args", func(t *testing.T) {
```

Comment:
do a test with a pipe, i.e. 
echo 'data' | decoder.go ... etc

Reply:
Pipe would not work. We need to read the output into a variable and then pass on the variable when decoder is used with the `cURL` as shown below:

 ```
curl \
     -H "Content-Type: application/json" \
     -H "UserID: bob" \
     -H "Signature: MEUCIQDm6dLmAdd0X49JygTiUkh+brZxprWSr2+hcAH+QIu3AAIgF+m7kO33YXyyqSbnXS9HR79wt/aL3JGhKvXFQaFBJms=" \
     -X GET http://127.0.0.1:6001/data/db2/key1 | read json; ./base64_decoder -getresponse=$json | jq .
```

Reply:
ok, that works